swap_meet/vendor.py

```diff
@@ -1,2 +1,160 @@
+    
+    ...
+    Methods:
+
+    add(new_item):
+        Adds provided item to instance inventory and returns the item. 
+
+    remove(discarded_item):
+        Removes provided item from instance inventory and returns the item.
+        - Returns False if item not already in inventory.
+
+    get_by_category(category_string):
+        Returns list of items of provided category from instance inventory.
+
+    swap_items(other, my_item, their_item):
+        Swaps provided items between two instances' inventories and returns True.
+        - Returns False if either item not found.
+
+    swap_first_item(other):
+        Swaps first items of two instances' inventories and returns True.
+        - Returns false if either inventory is empty.
+
+    get_best_by_category(desired_category):
+        Returns item of provided category with best condition from an instance's inventory. 
+        - Returns None if no items of provided category in inventory.
+
+    swap_best_by_category(other, my_priority, their_priority):
+        Swaps best items in provided categories between two instances' inventories and returns True.
+        - Returns False if either item not found. 
+
+    get_newest_item():
+        Returns newest item from instance's inventory items with known ages.
+        - Returns None if all items ages are unknown.
+
+    swap_by_newest(other):
+        Swaps newest items of two instances' inventories and returns True.
+        - Returns False if either or both inventory items' ages are all unknown.
+    
+    """
+
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, new_item):
+        """Add provided item to inventory"""
+
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, discarded_item):
+        """Remove an item from inventory"""
+
+        if discarded_item not in self.inventory:
+            return False
+        else: 
+            self.inventory.remove(discarded_item)
+            return discarded_item
+    
+    def get_by_category(self, category_string):
+        """Return list of items from inventory of a certain category"""
+
+        item_list = []
+        for item in self.inventory:
+            if item.category == category_string:
+                item_list.append(item)
+        return item_list
+    
+    def swap_items(self, other, my_item, their_item):
+        """Swaps items betwen self and another vendor's inventory"""
+
+        # check valid input
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+        else:
+            # remove items from original inventory
+            self.remove(my_item)
+            other.remove(their_item)
+            # add items to new inventory 
+            self.add(their_item)
+            other.add(my_item)
+            return True
```

Comment:
Similar to the feedback in the `remove` function, since the else-block of an if-statement is optional, we could remove the else and outdent the main logic of the function:
```suggestion
        if my_item not in self.inventory or their_item not in other.inventory:
            return False
            
        # remove items from original inventory
        self.remove(my_item)
        other.remove(their_item)
        # add items to new inventory 
        self.add(their_item)
        other.add(my_item)
        
        return True
```

Reply:
I think I'm just such a completionist that leaving the else bit off of an if else statement feels "wrong" still 😅. I'll work on it!